product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioSampleTest.java

```diff
@@ -0,0 +1,178 @@
+                Assert.assertEquals(response.getStatusLine().getStatusCode(), 200);
+                Assert.assertEquals(IOUtils.toString(response.getEntity().getContent()), responseString);
+            }
+        } catch (IOException e) {
+            throw e;
+        }
+    }
+
+    @Test(description = "1.1.1.2")
+    public void testMessageTransformationForInvalidCode() throws Exception {
+        // Invoke the service and invoke
+        String restURL = infraProperties.getProperty(ESB_HTTP_URL) + "/city/lookup/6060100";
+        HttpGet httpHead = new HttpGet(restURL);
+        try (CloseableHttpClient httpClient = HttpClientBuilder.create().setDefaultRequestConfig(requestConfig).build()) {
+            try (CloseableHttpResponse response = httpClient.execute(httpHead)) {
+                log.info(response.getStatusLine());
+                String responseString = "{\n" +
+                                        "  \"Error\": {\n" +
+                                        "    \"message\": \"Error while processing the request\",\n" +
+                                        "    \"code\": \"0\",\n" +
+                                        "    \"description\": \"Error while building message. Error while building Passthrough stream\"\n" +
+                                        "  }\n" +
+                                        "}";
+                Assert.assertEquals(IOUtils.toString(response.getEntity().getContent()), responseString);
+            }
+        } catch (IOException e) {
+            throw e;
+        }
+    }
+
+    @Test(description = "Test HTTP the transformation when a invalid status code is given", enabled = false)
+    public void testMessageTransformationFailure() throws Exception {
+        Assert.assertTrue(false, "This test is intentionally failed!");
+    }
+
+    @AfterClass(description = "Test HTTP the transformation")
+    public void close() throws Exception {
+        // Clean up if required
+    }
+
+    private boolean isCarFileDeployed(String carFileName) throws Exception {
+
+        log.info("waiting " + ARTIFACT_DEPLOYMENT_WAIT_TIME_MS + " millis for car deployment " + carFileName);
+        boolean isCarFileDeployed = false;
+        Calendar startTime = Calendar.getInstance();
+        long time;
+        while ((time = (Calendar.getInstance().getTimeInMillis() - startTime.getTimeInMillis())) <
+               ARTIFACT_DEPLOYMENT_WAIT_TIME_MS) {
+            String[] applicationList = applicationAdminClient.listAllApplications();
+            if (applicationList != null) {
+                if (ArrayUtils.contains(applicationList, carFileName)) {
+                    isCarFileDeployed = true;
+                    log.info("car file deployed in " + time + " mills");
+                    return isCarFileDeployed;
+                }
+            }
+            try {
+                Thread.sleep(1000);
+            } catch (InterruptedException e) {
+                //ignore
```

Comment:
We should not ignore InterruptedExceptions. 
https://rules.sonarsource.com/java/tag/multi-threading/RSPEC-2142

Reply:
This file has been removed by PR [https://github.com/wso2/product-ei/pull/2977](https://github.com/wso2/product-ei/pull/2977)